### PR TITLE
Add support for userSettingsValues to HubitatDeviceScript and Add DeviceSandboxTest

### PR DESCRIPTION
**Pull request by cmorse**

### CODE REVIEW

File: src/test/groovy/me/biocomp/hubitat_ci/device/DeviceSandboxTest.groovy [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    if (LoginCheck) { log.debug '1' }
+    else { log.debug '2' }
+} 
+""").run(noValidation: true)
+
+        then:
+            _*api.getLog() >> log
+    }
+
+    private def makeScriptForPrivateCheck(def fileOrText) {
```
Comment (biocomp) [inline src/test/groovy/me/biocomp/hubitat_ci/device/DeviceSandboxTest.groovy]: Try replacing use of `noValidation` with appropriate `validationFlags`. I'm not sure what it was doing - it was a while ago that I wrote that test :);

File: src/test/groovy/me/biocomp/hubitat_ci/device/DeviceSandboxTest.groovy [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+                    new Tuple("getProducedPreferences()", "getProducedPreferences"),
+                    new Tuple("void foo() { def prefs = producedPreferences }", "producedPreferences"),
+                    new Tuple("getProducedDefinition()", "getProducedDefinition"),
+                    new Tuple("void foo() { def prefs = producedDefinition }", "producedDefinition"),
+                    new Tuple("printf", "printf"),
+                    new Tuple("sleep 10", "sleep"),
+            ])
+    }
+
+    def "#description not allowed and should fail to compile"() {
```
Comment (biocomp) [inline src/test/groovy/me/biocomp/hubitat_ci/device/DeviceSandboxTest.groovy]: `@Unroll` is needed too

**Comment by cmorse** [comment]
I made all the changes you requested

**Comment by cmorse** [comment]
> Add test for `userSettingsValues` for `HubitatDeviceScript`.

Was this commend meant for me? I didn't notice it before.

**Comment by biocomp** [comment]
> 
> 
> > Add test for `userSettingsValues` for `HubitatDeviceScript`.
> 
> Was this commend meant for me? I didn't notice it before.

Yup. I did not feel like I need to hold the pull request over this, but if you feel like it, send another one with the test :)
Github code review UI is horrible, I did not know where to put this comment, and no wonder that you have not noticed it.

**Pull request merged**